icu4j/main/classes/core/src/com/ibm/icu/impl/number/UsagePrefsHandler.java

```diff
@@ -0,0 +1,110 @@
+// © 2020 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+package com.ibm.icu.impl.number;
+
+import com.ibm.icu.impl.IllegalIcuArgumentException;
+import com.ibm.icu.impl.units.MeasureUnitImpl;
+import com.ibm.icu.impl.units.UnitsRouter;
+import com.ibm.icu.number.Precision;
+import com.ibm.icu.util.Measure;
+import com.ibm.icu.util.MeasureUnit;
+import com.ibm.icu.util.ULocale;
+
+import java.math.BigDecimal;
+import java.math.MathContext;
+import java.util.ArrayList;
+import java.util.List;
+
+public class UsagePrefsHandler implements MicroPropsGenerator {
+
+private final MicroPropsGenerator fParent;
+private UnitsRouter fUnitsRouter;
+
+public UsagePrefsHandler(ULocale locale, MeasureUnit inputUnit, String usage, MicroPropsGenerator parent) {
+    this.fParent = parent;
+    this.fUnitsRouter =
+            new UnitsRouter(MeasureUnitImpl.forIdentifier(inputUnit.getIdentifier()), locale.getCountry(), usage);
+}
+
+private static Precision parseSkeletonToPrecision(String precisionSkeleton) {
+    final String kSuffixPrefix = "precision-increment/";
+    if (!precisionSkeleton.startsWith(kSuffixPrefix)) {
+        throw new IllegalIcuArgumentException("precisionSkeleton is only precision-increment");
+    }
+
+    String skeleton = precisionSkeleton.substring(kSuffixPrefix.length());
+    String skeletons[] = skeleton.split("/");
+    BigDecimal num = new BigDecimal(skeletons[0]);
+    BigDecimal den =
+            skeletons.length == 2 ?
+                    new BigDecimal(skeletons[1]) :
+                    new BigDecimal("1");
+
+
+    return Precision.increment(num.divide(den, MathContext.DECIMAL128));
+}
+
+protected static void mixedMeasuresToMicros(List<Measure> measures, DecimalQuantity quantity, MicroProps micros) {
+    if (measures.size() > 1) {
```

Comment:
If measures.size() is only 1, you're not setting quantity. I think this function must always set quantity, based on what came in via measures.

Reply:
@younies here's the issue you fixed with a44e0141e3582e008446bec67cf11b019f50b571 !